File: frontend/src/types/vocabularyDashboard.ts
export interface CategorySummary {
  id: string;
  name: string;
  type: string;
  hsk_level: number | null;
}

export interface VocabularyDashboardItem {
  vocabulary_id: string;
  text: string;
  pinyin: string;
  meaning: string;
  status: string;
  successful_recalls: number;
  failed_recalls: number;
  times_seen: number;
  last_practised_at: string | null;
  last_seen_at: string | null;
  categories: CategorySummary[];
  hsk_level: number | null;
}

export interface VocabularyDashboardResponse {
  items: VocabularyDashboardItem[];
  total: number;
  status_counts: Record<string, number>;
}

export interface CategoryListItem {
  id: string;
  name: string;
  type: string;
  parent_id: string | null;
  sort_order: number;
  hsk_level: number | null;
}

export interface CategoryListResponse {
  categories: CategoryListItem[];
}

export interface VocabularyDashboardFilters {
  status?: string | null;
  category_id?: string | null;
  hsk_level?: number | null;
  search?: string | null;
}

File: backend/src/chinese_learning/presentation/rest/schemas/vocabulary_dashboard.py
"""Schemas for the vocabulary dashboard API."""

from pydantic import BaseModel, Field


class CategorySummarySchema(BaseModel):
    id: str
    name: str
    type: str
    hsk_level: int | None = None


class VocabularyDashboardItemSchema(BaseModel):
    vocabulary_id: str
    text: str
    pinyin: str
    meaning: str
    status: str
    successful_recalls: int
    failed_recalls: int
    times_seen: int
    last_practised_at: str | None = None
    last_seen_at: str | None = None
    categories: list[CategorySummarySchema] = Field(default_factory=list)
    hsk_level: int | None = None
    characters: list[str] = Field(default_factory=list)


class VocabularyDashboardResponse(BaseModel):
    items: list[VocabularyDashboardItemSchema]
    total: int
    status_counts: dict[str, int]


class CategoryListItemSchema(BaseModel):
    id: str
    name: str
    type: str
    parent_id: str | None = None
    sort_order: int
    hsk_level: int | None = None


class CategoryListResponse(BaseModel):
    categories: list[CategoryListItemSchema]
